### PR TITLE
fix(sanitizer): remove orphaned tool_results before upstream

### PR DESCRIPTION
**Pull request by Bowl42**
## Summary
- Add `RemoveOrphanedToolResults()` to bedrock sanitizer — removes `tool_result` blocks whose `tool_use_id` has no matching `tool_use` in the preceding assistant message
- Called from both direct Bedrock adapter and custom adapter's `processClaudeRequestBody` pipeline
- Prevents Bedrock/Anthropic 400 errors (`unexpected tool_use_id found in tool_result blocks`) that previously cascaded through all provider channels

## Root cause
Clients sometimes send `tool_result` blocks referencing stale `tool_use_id`s from earlier turns. The API rejects these with a 400 error. Without sanitization, every provider channel fails on the same bad input.

## Test plan
- [x] 5 unit tests covering: orphan removal, all-orphaned fallback, no-op on valid input, no-op without tool_results, first-message edge case
- [x] All existing provider adapter tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Bug Fixes
* 优化了AI工具调用消息的处理流程，移除孤立的工具结果记录，确保消息链的完整性和一致性

## Tests
* 为工具结果处理功能添加了全面的测试套件

### CODE REVIEW

**Review by Bowl42 — COMMENTED** [review]
## Claude Review ✅

**`bedrock/sanitizer.go` — `RemoveOrphanedToolResults()`**: Clean implementation. The function correctly:
- Only processes user messages that follow an assistant message (skips index 0 and non-assistant predecessors)
- Collects valid `tool_use` IDs from the preceding assistant, then filters the user message's content blocks
- Falls back to a `[empty]` text block when all content is orphaned (avoiding empty content array rejection)
- Short-circuits with `len(validIDs) == 0` when the assistant has no tool_use blocks

**`sanitizeRequestBody()` integration**: Correctly placed after `SanitizeForBedrockCompat()` — the order matters because Bedrock compat may strip fields but won't affect tool_use/tool_result correspondence.

**`custom/claude_body.go` integration**: Correct ordering — `RemoveOrphanedToolResults` runs before `ensureToolResultCorrespondence`. Remove orphans first, then inject missing placeholders.

**Tests**: 5 cases covering core scenarios including edge cases (first message, all-orphaned, no tool_results).

No concerns. LGTM.

**Comment by Bowl42** [comment]
@coderabbitai review

File: internal/adapter/provider/bedrock/sanitizer.go [hunk 1]
```diff
@@ -113,6 +116,113 @@ func EnsureMaxTokensAboveThinkingBudget(body []byte) []byte {
...
+				}
+				return true
+			})
+		}
+
+		// If the preceding assistant has no tool_use blocks, nothing to orphan.
+		if len(validIDs) == 0 {
+			rebuilt = append(rebuilt, msg.Raw)
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (Bowl42) [reply internal/adapter/provider/bedrock/sanitizer.go]: Fixed. Removed the `len(validIDs) == 0` early return so orphaned tool_results are cleaned up even when the preceding assistant has only text. Added a regression test for this case.

**Review by awsl233777 — APPROVED** [review]
Checks are green and the orphaned tool_result sanitizer fix is covered by regression tests. Approving for merge.
